File: src/Entity/Event.php
<?php

namespace App\Entity;

use App\Repository\EventRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\HttpFoundation\File\File;

#[ORM\Entity(repositoryClass: EventRepository::class)]
class Event
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: "string", length: 255, nullable: true)]
    private ?string $image = null;

    private ?File $file = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "The name should not be null")]
    private ?string $name = null;

    #[ORM\Column(type: Types::TEXT)]
    #[Assert\NotBlank(message: "The description should not be null")]
    private ?string $description = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    #[Assert\GreaterThanOrEqual("today", message: "The start date cannot be in the past.")]
    private ?\DateTimeInterface $dateDebut = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    #[Assert\GreaterThanOrEqual(propertyPath: "dateDebut", message: "The end date cannot be before the start date.")]
    private ?\DateTimeInterface $dateFin = null;

    #[ORM\OneToMany(targetEntity: Participation::class, mappedBy: 'event')]
    private Collection $participations;

    public function __construct()
    {
        $this->participations = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }

    public function getFile(): ?File
    {
        return $this->file;
    }
 
    public function setFile(?File $file): self
{
    $this->file = $file;

    return $this;
}

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): self
    {
        $this->name = $name;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getDateDebut(): ?\DateTimeInterface
    {
        return $this->dateDebut;
    }

    public function setDateDebut(\DateTimeInterface $dateDebut): self
    {
        $this->dateDebut = $dateDebut;
        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->dateFin;
    }

    public function setDateFin(\DateTimeInterface $dateFin): self
    {
        $this->dateFin = $dateFin;
        return $this;
    }

    public function getParticipations(): Collection
    {
        return $this->participations;
    }

    public function addParticipation(Participation $participation): self
    {
        if (!$this->participations->contains($participation)) {
            $this->participations->add($participation);
            $participation->setEvent($this);
        }
        return $this;
    }

    public function removeParticipation(Participation $participation): self
    {
        if ($this->participations->removeElement($participation)) {
            if ($participation->getEvent() === $this) {
                $participation->setEvent(null);
            }
        }
        return $this;
    }
}

File: src/main/java/project/models/event.java
package project.models;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class event {
    private int ID;
    private String Name;
    private String date_debut;
    private String date_fin;
    private String Description;
    private String image;
    private List<participation> participations = new ArrayList<>();

    public event() {}

    public event(int ID, String name, String date_debut, String date_fin, String description, String image) {
        this.ID = ID;
        Name = name;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        Description = description;
        this.image = image;
    }
    public event(String name, String date_debut, String date_fin, String description, String image) {
        this.Name = name;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.Description = description;
        this.image = image;
    }

    // Getters & Setters
    public int getID() { return ID; }
    public void setID(int ID) { this.ID = ID; }
    public String getName() { return Name; }
    public void setName(String name) { Name = name; }
    public String getDate_debut() { return date_debut; }
    public void setDate_debut(String date_debut) { this.date_debut = date_debut; }
    public String getDate_fin() { return date_fin; }
    public void setDate_fin(String date_fin) { this.date_fin = date_fin; }
    public String getDescription() { return Description; }
    public void setDescription(String description) { Description = description; }
    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }

    // Gestion des participations
    public List<participation> getParticipations() { return participations; }
    public void setParticipations(List<participation> participations) { this.participations = participations; }
    public void addParticipation(participation p) { this.participations.add(p); }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        event event = (event) o;
        return ID == event.ID &&
                Objects.equals(Name, event.Name) &&
                Objects.equals(date_debut, event.date_debut) &&
                Objects.equals(date_fin, event.date_fin) &&
                Objects.equals(Description, event.Description) &&
                Objects.equals(image, event.image);
    }

    @Override
    public int hashCode() {
        return Objects.hash(ID, Name, date_debut, date_fin, Description, image);
    }

    @Override
    public String toString() {
        return "event{" +
                "ID=" + ID +
                ", Name='" + Name + '\'' +
                ", date_debut='" + date_debut + '\'' +
                ", date_fin='" + date_fin + '\'' +
                ", Description='" + Description + '\'' +
                ", image='" + image + '\'' +
                ", participations=" + participations.size() +
                '}';
    }
}